### ISSUE TITLE
Simplify and diversify node setup instructions

### ISSUE THREAD

**Issue by whoabuddy** [blockstack/docs#930]
## TL;DR

Setting up a node can take a few different routes depending on the user, their configuration, and their experience level. Right now, some of the instructions are not as clear as they could be, and could benefit from reorganization based on the desired user journey.

## Details

Running a node feels very technical and somewhat intimidating for the average user, mining doubly so.

The tutorial is targeted at building the code when binaries are available for download [as releases](https://github.com/blockstack/stacks-blockchain/releases/latest), and the instructions are not consistent between operating systems. 

To help sort this out, I came up with the following main questions or pathways that would affect the steps a user would take and the prerequisites required to get their node up and running.

The main questions/pathways I came up with are:
- what is your target OS/environment? *(Windows/Mac/Linux/Docker/Pi/etc)*
- do you want to run a follower or a miner?
- do you want to build the code or download a release?
- do you want to view output in the terminal or the browser? *(latter point to Mining-Bot)*
- do you want to connect to testnet or mainnet? *(forward-thinking)*

I'm also working on a flowchart graphic to better illustrate this, draft copy below:

![20201123_001_Selection](https://user-images.githubusercontent.com/9038904/100122568-c26d8080-2e36-11eb-946a-60c2bb6dd228.png)

I just finished four one-on-one sessions setting up miners for users, and we ran into some interesting snags:
- building the code on MacOS requires Xcode Command Line Tools, which is not listed as a pre-req (same for git)
- running the binaries on MacOS results in an [unidentified developer](https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/mac) warning, which requires changing security settings to be able to run it
    - on one, we had to change the setting under Settings > General
    - on another, we had to ctrl+click the file to open it manually, or it was blocked by default
- one Mac was out of date and the curl command to install Rust didn't work, had to remove the `--proto '=https' --tlsv1.2` options
- saving the config file in Windows resulted in an extra `.txt` in the file extension, which is not seen if the default setting `Hide file extensions for known file types` is enabled
- to stop a running miner is `ctrl + c` on Windows/Linux and `command + .` on a Mac (took some trial and error to figure that second one out)

On average, each session took at least 30 minutes to get the miner up and running, and I'm still reviewing them for common pitfalls and ways to make a better experience. Each session easily rolled into 30 minutes of open discussion about mining, the mechanics behind it, how to check balances, and what the different outputs met.

## Simplest Approach

If I were to distill the steps down into the simplest form, it'd be:

1. Download latest release from GitHub
https://github.com/blockstack/stacks-blockchain/releases/latest

2. Generate keychain file (requires node.js)
`npx @stacks/cli make_keychain -t > keychain.json`

3. Download config file (from my repo - can copy/paste from the Stacks instructions, too)
`curl -sSO https://raw.githubusercontent.com/AbsorbingChaos/bks-setup-miner/master/krypton-miner-conf.toml`

4. Insert private key into config file (on the line `seed =`)

5. Get test Bitcoin from the faucet
`curl -XPOST "https://stacks-node-api.blockstack.org/extended/v1/faucets/btc?address=YOURBTCADDRESS"`
or via URL: https://www.blockstack.org/testnet/faucet

6. Start the miner!
Mac/Linux: `./stacks-node start --config=krypton-miner-conf.toml` (not in PATH by default)
Windows: `stacks-node.exe start --config=krypton-miner-conf.toml`

## Helpful links and commands

View saved keychain info:
Mac/Linux: `cat keychain.json`
Windows: `type keychain.json`

Check BTC balance:
https://stacks-node-api.krypton.blockstack.org/extended/v1/faucets/btc/BTCADDRESS

Check STX balance: 
http://testnet-miner.blockstack.org:20443/v2/accounts/STXADDRESS?proof=0
http://127.0.0.1:20443/v2/accounts/STXADDRESS?proof=0

Compare network status:
http://testnet-miner.blockstack.org:20443/v2/info
http://127.0.0.1:20443/v2/info
```
"burn_block_height":4713 <-- current bitcoin chain height
"stacks_tip_height":4314 <-- current stacks chain height
```

When using `curl` for json data, piping it into `json_pp` should work by default on Mac/Linux and makes it much easier to digest.
```bash
curl -sS "https://stacks-node-api.krypton.blockstack.org/extended/v1/faucets/btc/BTCADDRESS" | json_pp
curl -sS "http://testnet-miner.blockstack.org:20443/v2/accounts/STXADDRESS?proof=0" | json_pp
cat keychain.json | json_pp
```

**When in doubt, [read the SIPs!](https://github.com/blockstack/stacks-blockchain/tree/master/sip)**

## References

- https://twitter.com/PatrickWStanley/status/1331092446608932864
- https://www.youtube.com/watch?v=N787Ok3l4qs
- https://www.youtube.com/watch?v=Djb5Jxbjl44

**Comment by HariniRajan397** [comment]
This is awesome. I agree simplifying things and diversifying instructions will be helpful for newcomers. We can even add a video tutorial link for each setup. Making a list of helpful commands is really useful. On the same lines, I raised a PR https://github.com/blockstack/docs/issues/881 as well during Docs week. Overall, wonderful effort, Jason @whoabuddy !!

**Comment by agraebe** [comment]
I love this, thanks @whoabuddy! please let us know how we can support this effort. I'm assuming this will ultimately end up in a list of content changes and potential docs system changes (code sample toggle for different environments, conditional steps, etc), is that correct?

while you're working on the new approach, are there some low hanging fruits we can work on together (ie smaller content changes)?



> This is awesome. I agree simplifying things and diversifying instructions will be helpful for newcomers. We can even add a video tutorial link for each setup.

absolutely love the idea of video instructions!
